unit_tests/test_cluster.py

```diff
@@ -507,6 +508,33 @@ def test_datacenter_name_per_region(self):  # pylint: disable=no-self-use
         datacenter_name_per_region = db_cluster.get_datacenter_name_per_region(db_nodes=[node1])
         assert datacenter_name_per_region == {'east-us': 'eastus'}
 
+    def test_get_rack_names(self):  # pylint: disable=no-self-use
+        resp = "\n".join(["Datacenter: eastus",
+                          "==================",
+                          "Status=Up/Down",
+                          "|/ State=Normal/Leaving/Joining/Moving",
+                          "--  Address   Load       Tokens       Owns    Host ID                               Rack",
+                          "UN  10.0.59.34    21.71 GB   256          ?       e5bcb094-e4de-43aa-8dc9-b1bf74b3b346  1a",
+                          "UN  10.0.198.153  ?          256          ?       fba174cd-917a-40f6-ab62-cc58efaaf301  1a",
+                          "Datacenter: westus",
+                          "==================",
+                          "Status=Up/Down",
+                          "|/ State=Normal/Leaving/Joining/Moving",
+                          "--  Address   Load       Tokens       Owns    Host ID                               Rack",
+                          "UN  10.1.59.34    21.71 GB   256          ?       e5bcb094-e4de-43aa-8dc9-b1bf74546346  2a"
+                          ]
+                         )
+        node1 = NodetoolDummyNode(resp=resp, myregion="east-us", myname="10.0.59.34", myrack=1)
+        node2 = NodetoolDummyNode(resp=resp, myregion="east-us", myname="10.0.198.153", myrack=1)
+        node3 = NodetoolDummyNode(resp=resp, myregion="west-us", myname='10.1.59.34', myrack=2)
```

Comment:
IIUC to be precise, in such configuration, all the nodes should have `rack=0` - because in single rack scenarios we use only this index.
You could extend this test and make second DC to have 2 racks (e.g.:
```node4 = NodetoolDummyNode(resp=resp, myregion="west-us", myname='10.1.59.34', myrack=1)
[...]
assert datacenter_name_per_region == {('east-us', '1'): '1a', ('west-us', '1'): '2a', ('west-us', '2'): '2a'}
```

btw. `datacenter_name_per_region` rename to `rack_names_per_datacenter_and_rack_idx` or smthing like this

Reply:
changed the name, I don't think we need more unittest, the cases are the same, and it would work.

anyhow I would limit it to case of that we have more then on rack